File: java/src/test/java/problems201_300/ReverseLinkedListTest.java
package problems201_300;

import concepts.SinglyListNode;
import org.junit.Test;
import sample.SampledTest;

import static com.google.common.truth.Truth.assertWithMessage;

public class ReverseLinkedListTest extends SampledTest {
    @Test
    public void test() {
        for (ReverseLinkedList solution : ReverseLinkedList.values()) {
            for (ReverseLinkedListSample sample : getSamples(ReverseLinkedListSample[].class)) {
                assertWithMessage(sample.getMessage())
                    .that(solution.reverseList(SinglyListNode.of(sample.input.head)))
                    .isEqualTo(SinglyListNode.of(sample.output));
            }
        }
    }
}

File: kotlin/src/test/kotlin/problems201_300/ReverseLinkedListTest.kt
package problems201_300

import com.google.common.truth.Truth.assertWithMessage
import concepts.SinglyListNode
import sample.SampledTest
import sample.component1
import sample.component2
import sample.component3
import sample.getSamples
import kotlin.test.Test

class ReverseLinkedListTest : SampledTest() {
    @Test
    fun test() =
        ReverseLinkedList.entries.forEach { solution ->
            getSamples<Array<ReverseLinkedListSample>>().forEach { (input, output, message) ->
                assertWithMessage(message)
                    .that(solution.reverseList(SinglyListNode.of(*input.head)))
                    .isEqualTo(SinglyListNode.of(*output))
            }
        }
}
